R/rows.R

```diff
@@ -1,37 +1,50 @@
 #' Manipulate individual rows
 #'
 #' @description
-#' \lifecycle{experimental}
-#'
-#' These methods provide a framework for manipulating individual rows
-#' in existing tables, modeled after the SQL operations
-#' `INSERT`, `UPDATE` and `DELETE`.
-#' All operations expect existing and new data to be compatible.
+#' \Sexpr[results=rd, stage=render]{lifecycle::badge("experimental")}
+#'
+#' These functions provide a framework for modifying rows in table using
+#' a second table of data. The two tables are matched `by` a set of key
+#' variables whose values must uniquely identify each row. The functions are
+#' inspired by SQL's `INSERT`, `UPDATE` and `DELETE`, and can optionally
+#' modified `in_place` for selected backends.
+#'
+#' * `rows_insert()` adds new rows (like `INSERT`); key values in `y` must
+#'    not occur in `x`.
+#' * `rows_update()` modifies existing rows (like `UPDATE`); key values in
+#'   `y` must occur in `x`.
+#' * `rows_patch()` works like `rows_update()` but only overwrites `NA` values.
+#' * `rows_upsert()` inserts or updates depending on whether or not the the
+#'   key value in `x` exists in `y`.
```

Comment:
Is the other way 'round more natural?
```suggestion
#'   key value in `y` already exists in `x`.
```